File: library_zap/src/main/java/com/dolin/zap/util/LevelUtils.java
package com.dolin.zap.util;

import com.dolin.zap.Level;

/**
 * @author #Suyghur.
 * Created on 2021/05/19
 */
public class LevelUtils {

    public static String getLevelName(Level level) {
        return level.name();
    }

    public static String getShortLevelName(Level level) {
        String name;
        switch (level) {
            case NONE:
            case DEBUG:
                name = "D";
                break;
            case INFO:
                name = "I";
                break;
            case WARNING:
                name = "W";
                break;
            case ERROR:
                name = "E";
                break;
            default:
                if (level.ordinal() < Level.DEBUG.ordinal()) {
                    name = "D";
                } else {
                    name = "E";
                }
        }
        return name;
    }
}

File: library_zap/src/main/java/com/suyghur/dolin/zap/util/LevelUtils.kt
package com.suyghur.dolin.zap.util

import com.suyghur.dolin.zap.entity.Level

/**
 * @author #Suyghur.
 * Created on 4/8/21
 */
object LevelUtils {

    fun getLevelName(level: Level): String {
        return level.name
    }

    fun getShortLevelName(level: Level): String {
        return when (level) {
            Level.DEBUG -> "D"
            Level.INFO -> "I"
            Level.WARNING -> "W"
            Level.ERROR -> "E"
            else ->
                if (level.ordinal < Level.DEBUG.ordinal) {
                    "D"
                } else {
                    "E"
                }
        }
    }
}